ui/appui-react/src/appui-react/statusfields/MessageCenter.tsx

```diff
@@ -5,227 +5,165 @@
-                    this._changeActiveTab(MessageCenterActiveTab.AllMessages)
-                  }
-                >
-                  {UiFramework.translate("messageCenter.all")}
-                </MessageCenterTab>
-                <MessageCenterTab
-                  isActive={
-                    this.state.activeTab === MessageCenterActiveTab.Problems
-                  }
-                  onClick={() =>
-                    this._changeActiveTab(MessageCenterActiveTab.Problems)
-                  }
-                >
-                  {UiFramework.translate("messageCenter.errors")}
-                </MessageCenterTab>
-              </>
-            }
-            title={this._title}
-          >
-            {this.getMessages()}
-          </MessageCenterDialog>
-        </StatusBar.Popup>
-      </>
+  const isProblemStatus = (message: NotifyMessageDetailsType): boolean => {
+    // See priority values in DgnPlatform defined in NotificationManager
+    return (
+      message.priority === OutputMessagePriority.Error ||
+      message.priority === OutputMessagePriority.Fatal
     );
-
-    return footerMessages;
-  }
-
-  private _handleClose = () => {
-    this.setIsOpen(false);
   };
 
-  private _handleOutsideClick = (e: MouseEvent) => {
-    if (
-      !this._indicator.current ||
-      !(e.target instanceof Node) ||
-      this._indicator.current.contains(e.target)
-    )
-      return;
-
-    this._handleClose();
+  const notifyStatus = () => {
+    return messages.some((msg) => isProblemStatus(msg))
+      ? "negative"
+      : "primary";
   };
 
-  private _handleMessageIndicatorClick = () => {
-    this.setIsOpen(!this.state.isOpen);
-  };
-
-  private _changeActiveTab = (tab: MessageCenterActiveTab) => {
-    this.setState({ activeTab: tab });
+  const handleMessagesUpdatedEvent = () => {
+    // istanbul ignore else
+    setNotify(notifyStatus());
+    setMessages(MessageManager.messages);
   };
 
-  private getMessages(): React.ReactChild[] {
-    const messages = MessageManager.messages.slice(0).reverse();
-    const tabRows: React.ReactChild[] = new Array<React.ReactChild>();
+  React.useEffect(() => {
+    MessageManager.onMessagesUpdatedEvent.addListener(
+      handleMessagesUpdatedEvent
+    );
+    MessageManager.registerAnimateOutToElement(indicator.current);
 
-    messages.forEach((details: NotifyMessageDetailsType, index: number) => {
-      /* istanbul ignore else */
-      if (
-        this.state.activeTab === MessageCenterActiveTab.AllMessages ||
-        this.isProblemStatus(details.priority)
-      ) {
-        const iconClassName = MessageManager.getIconClassName(details);
-        const iconSpec = MessageManager.getIconSpecFromDetails(details);
-        const message = details.briefMessage;
+    return () => {
+      MessageManager.onMessagesUpdatedEvent.removeListener(
+        handleMessagesUpdatedEvent
+      );
+    };
+  });
+
+  const getMessages = (tab: string): React.JSX.Element[] => {
```

Comment:
Keep implicit return type.
```suggestion
  const getMessages = (tab: string) => {
```